examples/handle_quit.rs

```diff
@@ -0,0 +1,18 @@
+use macroquad::prelude::*;
+
+fn window_conf() -> Conf {
+    Conf {
+        window_title: "Handle quit event".to_owned(),
+        handle_quit_event: true,
```

Comment:
I would avoid tweaking miniquad's Conf for this. 

I think if we will allow "prevent_quit" call from macroquad - it will be just enough to cover all the cases

If window should not close on X click:
```
if quit_requested() {
  prevent_quit();
  
  draw_exit_dialog();
}
```


If window should close on first X click, but you want to get out of the loop:
```
if quit_requested() {
  break;
}
```

And default behaviour could be the same - macroquad will just exit on close event. I think to make it happen macroquad should exit not in "exit" handler, but somewhere near "frame" callback - to let quit prevention actually works